solr/solrj/src/test/org/apache/solr/client/solrj/impl/LBHttp2SolrClientIntegrationTest.java

```diff
@@ -337,8 +349,27 @@ public void startJetty() throws Exception {
         fail("TESTING FAILURE: could not grab requested port.");
       }
       this.port = newPort;
-      //      System.out.println("waiting.........");
-      //      Thread.sleep(5000);
+    }
+  }
+
+  private static class LBClientHolder implements AutoCloseable {
```

Comment:
Wouldn't closing lbClient close the delegate?  If it doesn't, shouldn't it?   Then LBClientHolder could go away.

Reply:
Well, looking at LBSolrClient#close (from 2018) -- from which all 3 implementations inherit -- it doesn't seem to.  And before I modified this test, it had both the inner and outer clients in the autoclosable try/catch.  I agree that this seems wrong and perhaps was an oversight long ago.  But then again, because we make the user create their own inner client as a separate step, perhaps it was thought it is the opener's responsibility to explicitly close.

Reply:
> we make the user create their own inner client as a separate step, perhaps it was thought it is the opener's responsibility to explicitly close

Ah; okay that makes sense; I didn't look previously.

Reply:
instead of a holder, could client return an anonymous subclass that propagates the close?

Reply:
The holder is just to avoid copy/paste in these test methods.  I'd rather test the actual class and not a subclass.